### PR TITLE
[CodePush] Initial structure and enums

### PR DESCRIPTION
**Pull request by annakocheshkova**

### CODE REVIEW

File: sdk/appcenter-codepush/proguard-rules.pro [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Add project specific ProGuard rules here.
+# By default, the flags in this file are appended to flags specified
+# in /usr/local/Cellar/android-sdk/24.3.3/tools/proguard/proguard-android.txt
```
Comment (MatkovIvan) [inline sdk/appcenter-codepush/proguard-rules.pro]: Some local path. Just remove the whole comment like in other modules

File: sdk/appcenter-codepush/build.gradle [hunk 2]
```diff
@@ -0,0 +1,7 @@
+project.description = 'This package contains functionalities to deploy your application in cloud and deliver updates for it.'
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/build.gradle]: I would prefer to replace this description with the following: "_A cloud service that enables developers to deploy mobile app updates directly to their users’ devices._" - https://github.com/Microsoft/code-push/

File: sdk/appcenter-codepush/build.gradle [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+evaluationDependsOn(':sdk')
+
+dependencies {
+    compile project(':sdk:appcenter')
+    compile "com.google.code.gson:gson:2.8.1"
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/build.gradle]: Gson version can be updated to 2.8.2 https://mvnrepository.com/artifact/com.google.code.gson/gson/2.8.2

File: sdk/appcenter-codepush/proguard-rules.pro [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+# If your project uses WebView with JS, uncomment the following
+# and specify the fully qualified class name to the JavaScript interface
+# class:
+#-keepclassmembers class fqcn.of.javascript.interface.for.webview {
+#   public *;
+#}
+
+# Invoked via reflection, when forcing javascript restarts.
+-keepclassmembers class com.facebook.react.ReactInstanceManagerImpl {
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/proguard-rules.pro]: It's RN specific, can we move it to the _AppCenter-SDK-React-Native_?

File: sdk/appcenter-codepush/proguard-rules.pro [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    void recreateReactContextInBackground();
+}
+
+-keepclassmembers class com.facebook.react.XReactInstanceManagerImpl {
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/proguard-rules.pro]: RN specific

File: sdk/appcenter-codepush/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+﻿<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.microsoft.appcenter.codepush">
+
+
+    <application>
+        <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/src/main/AndroidManifest.xml]: RN specific

File: sdk/appcenter-codepush/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+﻿<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.microsoft.appcenter.codepush">
+
+
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/src/main/AndroidManifest.xml]: Remove extra space

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package com.microsoft.appcenter.codepush.enums;
+
+import com.google.gson.annotations.SerializedName;
+
+/**
+ * A enum defining the state of an update.
+ */
+public enum CodePushUpdateState {
+
+    @SerializedName("0")
```
Comment (sergey-akhalkov) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java]: Would be great to add comments for all enum values, please see https://github.com/Microsoft/react-native-code-push/blob/master/typings/react-native-code-push.d.ts#L378 for details

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushSyncStatus.java [hunk 9]
```diff
@@ -0,0 +1,77 @@
...
+    /**
+     * The app is up-to-date with the CodePush server.
+     */
+    @SerializedName("0")
+    UP_TO_DATE(0),
+
+    /**
+     * An available update has been installed and will be run either immediately after the
+     * syncStatusChangedCallback function returns or the next time the app resumes/restarts,
+     * depending on the InstallMode specified in SyncOptions
```
Comment (max-mironov) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushSyncStatus.java]: missing dot here and below

Comment (MatkovIvan) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushSyncStatus.java]: Also, you can use `<code></code>` for code and `{@link }` for references

**Pull request merged**